python/freetensor/core/remote_task_scheduler.py

```diff
@@ -0,0 +1,779 @@
+from __future__ import annotations
+import queue
+import threading
+import time
+import copy
+from typing import Any, List
+from typing import Dict
+'''
+task_type is specified
+1.search_task
+2.measure_task
+(minus means result)
+special param:
+0.other(transfer control)
+then check param
+    trans_c:(those are conbined with additional param: time_stamp)
+    0. logger
+    1. broadcast availability
+    2. return inavailability
+    3. new_task_request    
+'''
+'''
+How does this part work:
+1. commit a task via methods like RemoteTaskScheduler.remote_measure_submit()
+    in this part, the task will be transformed into a Task object, then:
+        1.1 a unique threading lock is acquired
```

Comment:
AFAIK, `queue.Queue` is thread-safe. Do we really need a lock?

Reply:
Then why not modify the items *before* pushing them into the queue, or first popping them from the queue?

Reply:
Yes, That's what I am doing. But in this case, (I mean 1.1), Queue isn't used here. After merge is complete, the lock will be released. The lock is unique to each task, so it's just OK.

Reply:
What do you mean by "Queue isn't used here"? Is the lock per task or per queue?

Reply:
This is a threadlock for the task. So It will be locked at the time of registration and unlocked when the task is fully completed.